### ISSUE TITLE
ClientCredentials supporting EndpointParams

### ISSUE THREAD

**Issue by larryboymi** [Mastercard/terraform-provider-restapi#136]
I was trying to connect to an API recently which authenticated using OAuth but required an `audience` parameter to be set. I believe most handle this with [EndpointParams](https://github.com/golang/oauth2/blob/f6687ab2804c/clientcredentials/clientcredentials.go#L44) in the `clientcredentials` package, which I believe [you support](https://github.com/Mastercard/terraform-provider-restapi/blob/ec5248a628807e7332262b2d345a8c5c73885685/restapi/api_client.go#L158). 

I was just hoping that you could support passing through these parameters from the provider into the module.

Thanks!

**Comment by DRuggeri** [comment]
Hey, @larryboymi - that sounds like a straight forward enough thing to do. I started work on this, but got stymied early on with Terraform's Schema. I'm actually not sure if it's possible to say "I want to allow a map with arbitrary keys and an array of values here" in the provider dialog.

Sadly, I only got this far:
```
                                                "endpoint_params": {
                                                        Type:        schema.TypeList,
                                                        Optional:    true,
                                                        Description: "Additional values to pass to the underlying Oauth client library (as EndpointParams)",
                                                        Elem:        &schema.Resource{
                                                                Schema: map[string]*schema.Schema{
                                                                        "??????": {   // <--- How do we enable arbitrary keys???
                                                                                Type:        schema.TypeList,
                                                                                Elem:        &schema.Schema{Type: schema.TypeString},
                                                                                Optional:    true,
                                                                                Description: "An arbitrary set of values to pass for this key",
                                                                        },
                                                                }
                                                        }
                                                },
```

I'm game for adding this functionality because it should be really straight forward to build the EndpointParams (which are just [url.Values](https://pkg.go.dev/net/url#Values) easily constructed with strings.

Tagging this issue with help wanted to see if anyone has an idea?

**Comment by DRuggeri** [comment]
Nevermind the last reply - I wasn't caffeinated enough! I've implemented this in 298e1d56e4a61f2897f81ce499aaee255424efda!

**Comment by larryboymi** [comment]
> Nevermind the last reply - I wasn't caffeinated enough! I've implemented this in [298e1d5](https://github.com/Mastercard/terraform-provider-restapi/commit/298e1d56e4a61f2897f81ce499aaee255424efda)!

Thank you!!
